### PR TITLE
feat: reload explorer on buf enter

### PR DESCRIPTION
**Pull request by sarmong**
I found myself pressing `r` too often to update the tree.

I think this is a rather useful feature if you create/modify/delete files in the terminal. I don't really see downsides to this, but I can put it behind a config option if needed.

### CODE REVIEW

File: lua/nvim-tree.lua [hunk 1]
```diff
@@ -336,6 +336,8 @@ local function setup_autocommands(opts)
   if opts.hijack_directories.enable then
     create_nvim_tree_autocmd({ "BufEnter", "BufNewFile" }, { callback = M.open_on_directory })
   end
+
+  create_nvim_tree_autocmd("BufEnter", { pattern = "NvimTree_*", callback = reloaders.reload_explorer })
```
Comment (kyazdani42) [inline lua/nvim-tree.lua:340]: I wouldn't do that by default, some people have large trees and reloading is a bit expensive. This should be hidden behind a config option at least :) something like `reload_on_bufenter`, which should be falsy by default.

**Review by kyazdani42 — CHANGES_REQUESTED** [review]

File: lua/nvim-tree.lua (hunk 1 above)
Comment (alex-courtis) [reply lua/nvim-tree.lua:340]: Agreed.

That said, this is an option I will probably turn on, as I do find myself refreshing after external changes / make etc.

Comment (sarmong) [reply lua/nvim-tree.lua]: Added

File: doc/nvim-tree-lua.txt [hunk 2]
```diff
@@ -259,6 +259,10 @@ Keeps the cursor on the first letter of the filename when moving in the tree.
 Changes the tree root directory on `DirChanged` and refreshes the tree.
   Type: `boolean`, Default: `false`
 
+*nvim-tree.reload_on_bufenter
+Automatically reloads the tree on `BufEnter`.
```
Comment (alex-courtis) [inline doc/nvim-tree-lua.txt:263]: We can clarify:

Automatically reloads the tree on `BufEnter` nvim-tree.

File: README.md [hunk 3]
```diff
@@ -128,6 +128,7 @@ require'nvim-tree'.setup { -- BEGIN_DEFAULT_OPTS
   open_on_tab = false,
   sort_by = "name",
   update_cwd = false,
+  reload_on_bufenter = false,
```
Comment (alex-courtis) [inline README.md:131]: Please run `update-default-opts.sh`, see [CONTRIBUTING.md](https://github.com/kyazdani42/nvim-tree.lua/blob/master/CONTRIBUTING.md)

**Review by alex-courtis — CHANGES_REQUESTED** [review]

File: doc/nvim-tree-lua.txt [hunk 4]
```diff
@@ -259,6 +260,10 @@ Keeps the cursor on the first letter of the filename when moving in the tree.
 Changes the tree root directory on `DirChanged` and refreshes the tree.
   Type: `boolean`, Default: `false`
 
+*nvim-tree.reload_on_bufenter
```
Comment (kyazdani42) [inline doc/nvim-tree-lua.txt:263]: missing a `*` after nvim-tree.reload_on_bufenter

**Review by kyazdani42 — CHANGES_REQUESTED** [review]

File: doc/nvim-tree-lua.txt (hunk 4 above)
Comment (sarmong) [reply doc/nvim-tree-lua.txt]: fixed

**Review by kyazdani42 — APPROVED** [review]

**Review by alex-courtis — APPROVED** [review]
Works beautifully

**Pull request merged**